### ISSUE TITLE
Use different cross section data for testing

### ISSUE THREAD

**Issue by paulromano** [openmc-dev/openmc#1227]
The NNDC data that we use for testing on Travis only contains a single temperature for pointwise cross sections. This means that our regression testing currently misses temperature interpolation, which has caused problems in the past. The NNDC data used to be one of the only sets of openly available ACE files, but now we have many options, and we also have pregenerated HDF5 libraries (https://openmc.mcs.anl.gov). We should switch to using one of those so that we can increase coverage in regression testing.

**Comment by smharper** [comment]
We should also be mindful of the fact that many folks will see us as endorsing this cross section set---even if we only think it should be used for tests.  I think we should take the time to test the cross sections and make sure we're happy with the results before we "publish" them in the OpenMC repo.  We've got the cold criticality benchmarks, but it would also be nice to throw in a few others like Doppler coefficient and maybe foil activation.

**Comment by paulromano** [comment]
@smharper Yes, I agree we should do some integral testing on the cross sections before deciding to use them. @amandalund is currently working on sprucing up some automated benchmarking scripts that I had developed a while ago. Once those are ready, we'll be in a good position to compare different libraries.

A few more things to mention:
- We are also missing continuous S(a,b) in testing (the NNDC data does not use it)
- There are a few areas of the code that have been marked as "REMOVE THIS" but haven't been removed because doing so would change the random number stream (and consequently the test results). As long as we're changing cross sections, which will almost certainly change test results, we might as well get rid of these blocks.
